Tests/GeneratorEngineTests/EngineTests.swift

```diff
@@ -0,0 +1,152 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import class AsyncHTTPClient.HTTPClient
+@testable import GeneratorEngine
+import struct Logging.Logger
+import struct SystemPackage.FilePath
+import XCTest
+
+private let encoder = JSONEncoder()
+private let decoder = JSONDecoder()
+
+private extension FileSystem {
+  func read<V: Decodable>(_ path: FilePath, as: V.Type) async throws -> V {
+    let fileStream = try await self.read(path)
+    var bytes = [UInt8]()
+    for try await chunk in fileStream {
```

Comment:
I'd ask the user to provide a limit here and maybe default it to 10 MiB or something. Limitless memory ballooning is never a good API.